bytecomp/bytelink.ml

```diff
@@ -525,13 +525,13 @@ let link_bytecode_as_c tolink outfile with_main =
          (Marshal.to_string (Symtable.initial_global_table()) []);
        output_string outchan "\n};\n\n";
        (* The sections *)
-       let sections : (string * Obj.t) list =
-         [ Bytesections.Name.to_string SYMB,
-           Symtable.data_global_map();
-           Bytesections.Name.to_string PRIM,
-           Obj.repr(Symtable.data_primitive_names());
-           Bytesections.Name.to_string CRCS,
-           Obj.repr(extract_crc_interfaces()) ]
+       let sections : (string * Obj.t) array =
```

Comment:
Can't we use a `Meta.bytecode_sections` record directly ?

Reply:
I tried to keep this close to the original code, because the version with string labels is more robust to changes to the bytecode format (if you change the format without bumping magic numbers, then using `Meta.bytecode_sections` would crash as it assumes the same layout, whereas this format will do additional checks on section names).